src/libmongoc/src/mongoc/mongoc-server-description.c

```diff
@@ -96,8 +93,8 @@ mongoc_server_description_reset (mongoc_server_description_t *sd)
    sd->me = NULL;
    sd->current_primary = NULL;
    sd->set_version = MONGOC_NO_SET_VERSION;
-   bson_oid_copy_unsafe (&kObjectIdZero, &sd->election_id);
-   bson_oid_copy_unsafe (&kObjectIdZero, &sd->service_id);
+   mcommon_oid_set_zero (&sd->election_id);
```

Comment:
I asked on slack whether this should be public or private, and i was told private is better.

Reply:
Quoting [slack](https://mongodb.slack.com/archives/GMY98JS86/p1731519919090929?thread_ts=1731518280.367459&cid=GMY98JS86):

> I would likely argue to keep it as private API unless there is an expected use case for users.

I expect there may not be much of a need for users to create or work with zeroed ObjectIDs.

Reply:
I wasn't focused much on whether it was public/private, just that it was in mcommon instead of libbson. I suppose libmongoc technically cannot rely on private headers within libbson, correct?

Reply:
> cannot rely on private headers within libbson, correct?

libmongoc supports building against a system install of libbson. AFAIK there is no established pattern for libmongoc to use private API in libbson.